scripts/generate-development-package-tarballs.mjs

```diff
@@ -0,0 +1,67 @@
+/**
+ * Generate tarballs from the theatre packages and publish them.
+ */
+
+import os from 'os'
+import path from 'path'
+
+const packagesToPack = [
+  '@theatre/core',
+  '@theatre/studio',
+  '@theatre/dataverse',
+  '@theatre/react',
+  '@theatre/browser-bundles',
+  '@theatre/r3f',
+]
+
+/**
+ * Generate the tarball for a workspace and publish it
+ *
+ * @param {string} workspace - Name of the workspace
+ * @param {string} latestCommitHash - Abbreviated hash of the latest commit
+ * @param {{location: string, name: string}[]} workspacesListObjects - Objects containing the paths to the workspaces
+ */
+async function generateAndPublishTarball(
+  workspace,
+  latestCommitHash,
+  workspacesListObjects,
+) {
+  const tarballName = `${workspace
+    // Get rid of the "@"
+    .slice(1)
+    // Replace the "/" with "-"
+    .replace('/', '-')}-${latestCommitHash}.tgz`
+  const pathToTarball = path.join(
+    __dirname,
+    '..',
+    workspacesListObjects.filter((wp) => {
+      return wp.name === workspace
+    })[0].location,
+    tarballName,
+  )
+  await $`yarn workspace ${workspace} pack --filename ${tarballName}`
```

Comment:
I think (5/10) you need to temporarily override the version of each package to the desired version. See `assignVersions()` in `release.mjs`